### ISSUE TITLE
no-attribute-decorator: why is it considered a bad practice?

### ISSUE THREAD

**Issue by dev054** [angular-eslint/angular-eslint#237]
I wonder if we could get a better explanation on why it's considered a bad practice. Curently we have this description in the rule: https://github.com/angular-eslint/angular-eslint/blob/97ac0f18fb9ed65750dbe7d8c79fae900d98d04e/packages/eslint-plugin/src/rules/no-attribute-decorator.ts#L27

...however, for us, it's too vague. A link pointing it in the Angular Style Guide, like this: https://github.com/angular-eslint/angular-eslint/blob/b30fd0e9c8fe35b7cda8117131c2f836a5bcace0/packages/eslint-plugin/src/rules/no-host-metadata-property.ts#L21

...would be really helpful, to see if it's worth enabling this rule at our repository.

**Comment by JamesHenry** [comment]
Thanks @dev054 Would you like to submit a PR for this?

**Comment by dev054** [comment]
> Thanks @dev054 Would you like to submit a PR for this?

Hmm, in fact, I'm asking about the clarification on why this rule exists. What are the cons in using `@Attribute` and why is it considered a bad practice, because I didn't found it nowhere.

**Comment by rafaelss95** [comment]
That's really interesting. I tried to found the reasoning in Angular docs (more specifically [v12.angular.attribute-decorator](https://angular.io/api/core/AttributeDecorator), [v12.angular.styleguide](https://angular.io/guide/styleguide) and even at [v2.angular.styleguide](https://v2.angular.io/docs/ts/latest/guide/style-guide.html)) but i didn't find anything.
I also tried to see if I could find something in [PR that created this rule at the codelyzer repo](https://github.com/mgechev/codelyzer/pull/1), but again, no luck, there's nothing that isn't already in the current description/message.

TBH, I can't answer why this rule exists for myself. I can see cases where `attribute` is not intended to change, such as a simple string, and then someone might prefer to use `@Attribute` over `@Input`.

That being said, unless there's a good reason (documentation or maybe a use-case) explaining why we should prefer `@Attribute` over `@input` I'd vote for deprecate this rule :man_shrugging:

cc @JamesHenry
